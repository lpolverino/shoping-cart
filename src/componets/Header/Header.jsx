import React from 'react'
import LinksButton from '../LinksButton/LinksButton'

const Header = () => {
  return (
    <div>
        <nav>
            <LinksButton link ="/" name="Home"></LinksButton>
            <LinksButton link ="shop" name="Shop"></LinksButton>
        </nav>
        <div>
            <button data-testid = "shop-element">Cart</button>
        </div>
    </div>
  )
}

export default Header