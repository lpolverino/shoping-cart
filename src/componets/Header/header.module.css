.header{
    display: flex;
    justify-content: center;
    align-items: center;
    border: 3px solid black;
}
.navigation{
    display: flex;
    justify-content: space-between;
    gap: 2.5rem;
}

.cartConteiner{
  margin-left: auto;

}

.cart{
    margin-right: 1rem;
    background-color: rgb(255, 255, 255);
    padding: 10px;
    border-radius: 50%;
}
.shoppingCart{
    width: 16px;
    height: 16px;
    border: none;
}

a {
    display: inline-block;
    text-decoration:none
  }
  a:after {
    content: "";
    display: block;
    width: 0;
    border-bottom: 2px solid;
    margin: 0 auto;
    transition:all 0.3s linear 0s;
  }
 
  a:hover:after {
    width: 90%;
  }

  .title{
    padding: 10px;
    font-size: 2rem;
    margin-right: auto;
  }