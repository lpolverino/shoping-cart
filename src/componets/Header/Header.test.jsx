
import { render, screen } from "@testing-library/react";
import { Link,MemoryRouter as Router } from 'react-router-dom';
import Header from "./Header";
import { expect } from "vitest";

describe("Header component", () => {
  it("renders with two buttons and the cart icon", () => {
    render(<Router><Header /></Router>);
    expect(screen.getByRole("navigation"))
    expect(screen.getByTestId("shop-element"))
  });
});
