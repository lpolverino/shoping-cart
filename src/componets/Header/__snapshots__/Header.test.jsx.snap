// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > renders with two buttons and the cart icon 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <a
          href="/"
        >
          Home
        </a>
        <a
          href="/"
        >
          Shop
        </a>
        <div>
          <button
            data-testid="shop-element"
          >
            Cart
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/"
      >
        Shop
      </a>
      <div>
        <button
          data-testid="shop-element"
        >
          Cart
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
